AlicerceTests/Extensions/OptionalTests.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  OptionalTests.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import XCTest
+
+@testable import Alicerce
+
+final class OptionalTests: XCTestCase {
+
+    func testThen_UsingNonOptionalValue_ShouldUnwrapTheValue() {
+        let anOptionalString: String? = "😎"
+
+        anOptionalString.then { value in
+            XCTAssert(true, "🚀 \(value) unwrapped ✅")
+        }
+    }
+
+    func testThen_UsingOptionalValue_ShoudNotUnwrapTheValue() {
+        let nullValue: String? = nil
+
+        nullValue.then { _ in
+            XCTAssert(false, "💥 nil unwrapped 😱")
+        }
+
+        XCTAssert(true, "🚀 value not unwrapped ✅")
```

Comment:
Same as above (assert always passes, and we can add a check using the `nullValue`'s own value)